mutiny-core/src/lib.rs

```diff
@@ -359,6 +418,128 @@ impl<S: MutinyStorage> MutinyWallet<S> {
+    ///   amount is too high". This means that the LSP cannot support the amount that the user
+    ///   requested. The user should request a smaller amount from the LSP.
+    ///
+    /// - [`MutinyError::LspConnectionError`]: Returned if the LSP server returns an error with
+    ///   a status of 500, indicating an "Internal Server Error", and a message that starts with
+    ///   "Failed to connect to peer". This means that the LSP is not connected to our node.
+    ///
+    /// If the server returns a status of 500 with a different error message,
+    /// a [`MutinyError::LspGenericError`] is returned.
+    pub async fn create_bip21(
+        &self,
+        amount: Option<u64>,
+        labels: Vec<String>,
+    ) -> Result<MutinyBip21RawMaterials, MutinyError> {
+        let invoice = if self.config.safe_mode {
+            None
+        } else {
+            // Check if a federation exists
+            let federation_ids = self.list_federations().await?;
+            if !federation_ids.is_empty() {
+                // Use the first federation for simplicity
+                let federation_id = &federation_ids[0];
+                let fedimint_client = self.fedimints.lock().await.get(federation_id).cloned();
+
+                match fedimint_client {
+                    Some(client) => {
+                        // Try to create an invoice using the federation
+                        match client.get_invoice(amount.unwrap_or_default()).await {
+                            Ok(inv) => Some(inv.bolt11.ok_or(MutinyError::WalletOperationFailed)?),
+                            Err(_) => None, // Handle the error or fallback to node_manager invoice creation
+                        }
+                    }
+                    None => None, // No federation client found, fallback to node_manager invoice creation
+                }
+            } else {
+                // Fallback to node_manager invoice creation if no federation is found
+                let inv = self
+                    .node_manager
+                    .create_invoice(amount, labels.clone())
+                    .await?;
+                Some(inv.bolt11.ok_or(MutinyError::WalletOperationFailed)?)
+            }
+        };
+
+        let Ok(address) = self.node_manager.get_new_address(labels.clone()) else {
+            return Err(MutinyError::WalletOperationFailed);
+        };
+
+        Ok(MutinyBip21RawMaterials {
+            address,
+            invoice,
+            btc_amount: amount.map(|amount| bitcoin::Amount::from_sat(amount).to_btc().to_string()),
+            labels,
+        })
+    }
+
+    /// Gets the current balance of the wallet.
+    /// This includes both on-chain and lightning funds.
+    ///
+    /// This will not include any funds in an unconfirmed lightning channel.
+    pub async fn get_balance(&self) -> Result<MutinyBalance, MutinyError> {
+        Ok(MutinyBalance::new(self.node_manager.get_balance().await?))
```

Comment:
fedimint balance should be added

Reply:
No fedimint funds exist at this stage in the commit but yes federation balances have been added. I'd rather not worry about the details of multi federation yet.